src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -72,26 +72,66 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
-                        existingObservation = await GetObservationFromServerAsync(identifier).ConfigureAwait(false);
-                    })
-                    .ExecuteAndCaptureAsync(async () =>
-                    {
-                        var mergedObservation = MergeObservation(config, existingObservation, observationGroup);
-                        return await _client.UpdateAsync(mergedObservation, versionAware: true).ConfigureAwait(false);
-                    }).ConfigureAwait(false);
+                     .Handle<FhirOperationException>(ex => ex.Status == System.Net.HttpStatusCode.Conflict || ex.Status == System.Net.HttpStatusCode.PreconditionFailed)
+                     .RetryAsync(2, async (polyRes, attempt) =>
+                     {
+                         // 409 Conflict or 412 Precondition Failed can occur if the Observation.meta.versionId does not match the update request.
+                         // This can happen if 2 independent processes are updating the same Observation simultaneously.
+                         // or
+                         // The update operation failed because the Observation no longer exists.
+                         // This can happen if a cached Observation was deleted from the FHIR Server.
+
+                         // Attempt to get the most recent version of the Observation.
+                         existingObservation = await GetObservationFromServerAsync(identifier).ConfigureAwait(false);
+
+                         // If the Observation no longer exists on the FHIR Server, it was most likely deleted.
+                         if (existingObservation == null)
+                         {
+                             // Remove the Observation from the cache (this version no longer exists on the FHIR Server.
+                             _observationCache.Remove(cacheKey);
+
+                             // Create a new Observation.
+                             result = await _client.CreateAsync(mergedObservation).ConfigureAwait(false);
+                         }
+                     })
+                     .ExecuteAndCaptureAsync(async () =>
+                     {
+                         if (result != null)
+                         {
+                             // If result is not null, this means that a cached Observation was deleted from the FHIR Server.
+                             // The RetryAsync has handled this condition, so no further action is required, return the result.
+                             operationType = ResourceOperation.Created;
+                             return result;
+                         }
+
+                         // Merge the new data with the existing Observation.
+                         mergedObservation = MergeObservation(config, existingObservation, observationGroup);
+
+                         // Check to see if there are any changes after merging.
+                         if (mergedObservation.IsExactly(existingObservation))

```

Comment:
>IsExactly [](http://example.com/codeflow?start=47&length=9)

This is part of the HL7.Fhir.Core library.  It looks like it does a deep compare on each element.  Fine using this for now but wonder if a alternative where we generate the raw json and hash it and compare might be better in the long run.